components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt

```diff
@@ -44,7 +44,21 @@ fun String.isExtensionUrl() = this.startsWith("moz-extension://")
  */
 fun String.isResourceUrl() = this.startsWith("resource://")
 
-fun String.toNormalizedUrl() = URLStringUtils.toNormalizedURL(this)
+/**
+ * Appends `http` scheme if no scheme is present in this String.
+ */
+fun String.toNormalizedUrl(): String {
+    val s = this.trim()
+    // Most commonly we'll encounter http or https schemes.
+    // For these, avoid running through toNormalizedURL as an optimization.
+    return if (!s.startsWith("http://", ignoreCase = true) &&
+        !s.startsWith("https://", ignoreCase = true)
```

Comment:
nit: Do we need any other protocols? An alternative could be using `Uri.parse()` and normalizing if there's no scheme at all and otherwise let it through as-is? But I can't think of any real life use cases.

Reply:
I misunderstood the code. Christian explained it to me. :)